pulsar/consumer_regex.go

```diff
@@ -320,6 +319,25 @@ func (c *regexConsumer) closed() bool {
 }
 
 func (c *regexConsumer) monitor() {
+	defer close(c.subscribeCh)
+	defer close(c.unsubscribeCh)
+
+	go func() {
+		for topics := range c.subscribeCh {
+			if len(topics) > 0 && !c.closed() {
+				c.subscribe(topics, c.dlq, c.rlq)
+			}
+		}
+	}()
+
+	go func() {
+		for topics := range c.unsubscribeCh {
+			if len(topics) > 0 && !c.closed() {
+				c.unsubscribe(topics)
+			}
+		}
+	}()
```

Comment:
What about combining these two channels into one single channel?
Otherwise, under poor network conditions, where the process of subscribing or unsubscribing may take an long period, scenarios like the following might occur:

- A consumer is created.
- Topic A is created.
- The consumer discovers the topic and starts to subscribe to Topic A.
- Topic A is deleted.
- The consumer discovers the topic again, but the knownTopics list remains unchanged. Thus, the consumer doesn't unsubscribe from Topic A.
- Topic A is subscribed to, but this action fails.

This chain of events leads to the final outcome: Topic A should be unsubscribed from, but it isn't.

Another scenario that could result in data loss:
- A consumer is created and subscribes to Topic A.
- Topic A is deleted.
- The consumer discovers the topic and begins to unsubscribe from Topic A.
- Topic A is recreated.
- The consumer discovers the topic again, but the knownTopics list remains unchanged. As a result, the consumer doesn't resubscribe to Topic A.
- Topic A remains unsubscribed.
The final outcome in this case: Topic A is not subscribed to, and any subsequent messages in Topic A will be lost.

Merging the channels could simplify this logic and make it more robust. It would be more effective to have the monitoring of this topics list occur on a single thread. This approach is guaranteed in the Java client, preventing the aforementioned issues from occurring.

Reply:
That's a good point. I would go one step further and simplify this logic to not use channels but rather process subscriptions in the thread that processes the ticker. What do you think?

Theres no point to the channels as there's a single producer and a single consumer. And this already runs in a separate goroutine specific to the auto discovery of topics. If in the future we have additional producers then channels can be reintroduced with any new requirements in mind.

Reply:
Great. That makes sense to me.

Reply:
Perfect, done in the latest commit.